### PR TITLE
feat: add java-backport releaser

### PR DESCRIPTION
**Pull request by chingor13**
This release strategy always bumps the patch version. We will use it for backporting fixes to `<major>.<minor>.x` branches.

### CODE REVIEW

File: test/releasers/java-backport.ts [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+        readFileSync(
+          resolve('./test/fixtures', 'latest-tag-sp-version.json'),
+          'utf8'
+        )
+      );
+      req.post('/graphql').reply(200, {
+        data: graphql,
+      });
+      const latestTag = await releasePR.latestTag(undefined, true);
+      expect(latestTag!.version).to.equal('1.127.1-sp.1-SNAPSHOT');
```
Comment (suztomo) [inline test/releasers/java-backport.ts:596]: Can we use a version without "-sp.1" for test data?

File: test/releasers/java-backport.ts [hunk 2]
```diff
@@ -0,0 +1,630 @@
...
+        readFileSync(
+          resolve('./test/fixtures', 'latest-tag-sp-version.json'),
+          'utf8'
+        )
+      );
+      req.post('/graphql').reply(200, {
+        data: graphql,
+      });
+      const latestTag = await releasePR.latestTag();
+      expect(latestTag!.version).to.equal('1.127.0-sp.1');
```
Comment (suztomo) [inline test/releasers/java-backport.ts:581]: Can we use a version without "-sp.1" for test data?

File: test/releasers/java-backport.ts [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+    // Indicates that there are no PRs currently waiting to be released:
+    sandbox
+      .stub(releasePR.gh, 'findMergedReleasePR')
+      .returns(Promise.resolve(undefined));
+
+    sandbox.stub(releasePR, 'latestTag').returns(
+      Promise.resolve({
+        name: 'v0.20.3',
+        sha: 'abc123',
+        version: '0.20.3',
```
Comment (suztomo) [inline test/releasers/java-backport.ts:67]: Can I expect an assertion in which this "0.20.3" becomes "0.20.4" after applying JavaBackport strategy? (If this is not the point of this test case, then never mind)

**Review by bcoe — APPROVED** [review]
Can we potentially retire this once we split out bumping logic from release logic this coming quarter?

File: test/releasers/java-backport.ts (hunk 2 above)
Comment (chingor13) [reply test/releasers/java-backport.ts:581]: I will just remove this test - it was copied from the LTS test. The purpose of this one was to detect the special -sp.1 version scheme which is irrelevant to this releaser.

File: test/releasers/java-backport.ts (hunk 3 above)
Comment (chingor13) [reply test/releasers/java-backport.ts:67]: For these tests, it's captured in the snapshot file (which records the data sent to the api calls).

**Comment by chingor13** [comment]
> Can we potentially retire this once we split out bumping logic from release logic this coming quarter?

Yes, Java will be the strategy for knowing what files to update and the versioning strategy would be something like "Always bump patch"

**Review by suztomo — APPROVED** [review]

File: __snapshots__/java-backport.js [hunk 4]
```diff
@@ -0,0 +1,2338 @@
+exports['JavaBackport creates a release PR against a feature branch: changes'] = `
+
+filename: CHANGELOG.md
+# Changelog
+
+### [0.20.4](https://www.github.com/googleapis/java-trace/compare/0.20.3...v0.20.4) (1983-10-10)
```
Comment (suztomo) [inline __snapshots__/java-backport.js:6]: I see 0.20.4 here. Thank you.

**Review by ralphstone1 — COMMENTED** [review]
- Okay

**Pull request merged**